src/solution.rs

```diff
@@ -0,0 +1,97 @@
+use std::process::Command;
+use anyhow::{anyhow, Result};
+use crate::Clash;
+use crate::outputstyle::OutputStyle;
+
+mod suite_run;
+use suite_run::SuiteRun;
+mod test_run;
+use test_run::TestRun;
+
+use self::test_run::TestRunResult;
+
+pub struct Solution {
+    clash: Clash,
+    build_command: String,
+    run_command: String,
+}
+
+impl Solution {
+    pub fn new(clash: Clash, build_command: String, run_command: String) -> Self {
+        Self { clash, build_command, run_command }
+    }
```

Comment:
This is actually what I spent most of my time thinking about since I'm not very familiar with Rust and its really difficult to make short files and functions on it.

Build could be an option I don't mind that.

However I wanted to remove as any responsibilities as I could from main since in its current state it is too much of a mess. My idea was to in fact make build and run separate but also have run automatically call build if it hasn't been run with some sort of flag, but I could not see a situation where a solution is built but not run.

I think I was wrong to remove this responsibility from main by moving it to solution, I think another file should be added that links them together.

Reply:
Changed `build_command` to use `Option<String>`, my brain is still writing ruby. I'll work on the rest of the feedback about turning it into an `Option<Command>`.